src/doc/rustc/src/platform-support/fuchsia.md

```diff
@@ -641,8 +641,60 @@ available on the [Fuchsia devsite].
 
 ### Running the compiler test suite
 
-Running the Rust test suite on Fuchsia is [not currently supported], but work is
-underway to enable it.
+Pre-requisites for running the Rust test suite on Fuchsia are:
+1. Installation of `rustc` via "[Targeting Fuchsia with a compiler built from source](#targeting-fuchsia-with-a-compiler-built-from-source)", including `config-env.sh` and `config.toml`.
+1. Checkout of Rust source.
+1. Download of the Fuchsia SDK. Minimum supported SDK version is [4xjxrGUrDbQ6_zJwj6cDN1IbWsWV5aCQXC_zO_Hu0XkC](https://chrome-infra-packages.appspot.com/p/fuchsia/sdk/core/linux-amd64/+/4xjxrGUrDbQ6_zJwj6cDN1IbWsWV5aCQXC_zO_Hu0XkC)
+
+Interfacing with the Fuchsia emulator is handled by our test runner script located
+at `${RUST_SRC_PATH}/src/ci/docker/scripts/fuchsia-test-runner.py`.
+
+We start by activating our Fuchsia test environment. From a terminal:
+
+**Issue command from ${RUST_SRC_PATH}**
+```sh
+src/ci/docker/scripts/fuchsia-test-runner.py start
+    --rust .
+    --sdk ${SDK_PATH}
+    --target-arch {x64,arm64}
+```
+
+Next, for ease of commands, we copy `config-env.sh` and `config.toml` into our Rust source
+code path, `${RUST_SRC_PATH}`.
+
+From there, we utilize `x.py` to run our tests, using the test runner script to
+run the tests on our emulator. To run the full `src/test/ui` test suite:
+
+**Run from ${RUST_SRC_PATH}**
+```sh
+( \
+    source config-env.sh &&                                                   \
+    ./x.py                                                                    \
+    --config config.toml                                                      \
+    --stage=2                                                                 \
+    test src/test/ui                                                          \
+    --target x86_64-fuchsia                                                   \
+    --run=always --jobs 1                                                     \
+    --test-args --target-rustcflags -L                                        \
```

Comment:
Does this work? As I recall, `--test-args` takes a single arg or a space separated list of args. But you would have to escape it to get the second behavior, e.g.

```
--test-args "--target-rustcflags -L"
--test-args "--target-rustcflags ${SDK_PATH}/arch/{x64|arm64}/sysroot/lib
```

What's going on here is you're passing an arg to pass to `compiletest` which is then specifying an arg to pass to `rustc`. So `compiletest` should see something like

`--target-rustcflags -L --target-rustcflags ${SDK_PATH}/arch/{x64,arm64}/sysroot/lib`

while `rustc` sees

`-L ${SDK_PATH}/arch/{x64,arm64}/sysroot/lib`

It's pretty gross :)

Reply:
This works, and is the only way to pass multiple `--test-args` in our current CI on Fuchsia. When using `--test-args ""`, the double-quotes confuse the arg parsing and this is the only workaround

We could make upstream have a nicer-looking double-quote, but this will bring it out of line with our CI on Fuchsia

Reply:
Understood on the double quotes not working, but without them I would think you need

```suggestion
    --test-args --target-rustcflags                                           \
    --test-args -L
```

If this somehow works I'm interested to know how, not that that needs to block this PR.

Reply:
Oh no -- I apologize! I completely misunderstood what you were saying -- yes, you are right on the need to add an additional `--test-args`. Creating a PR to fix now